sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -155,8 +156,9 @@ private[sql] class CacheManager extends Logging {
    * function will over invalidate.
    */
   private[sql] def invalidateCache(plan: LogicalPlan): Unit = writeLock {
+    val canonicalized = plan.canonicalized
     cachedData.foreach {
-      case data if data.plan.collect { case p if p.sameResult(plan) => p }.nonEmpty =>
+      case data if data.plan.collect { case p if p.sameResult(canonicalized) => p }.nonEmpty =>
```

Comment:
Hmm, yes, we should for now, but we can't force to use canonicalized plan and it is difficult to check all implementations so I thought `CacheManager` should canonicalize plans.